runtime/makelib/targets.mk.linux.inc.ftl

```diff
@@ -348,6 +352,15 @@ endif
       PPC_GCC_CXXFLAGS += -DLINUXPPC -m32 -fstack-protector
     endif
   endif
+<#elseif uma.spec.processor.riscv64>
+  CFLAGS   += -DRISCV64 -Wno-error=stringop-truncation -Wno-error=stringop-overflow
+  CXXFLAGS += -DRISCV64 -Wno-error=stringop-truncation -Wno-error=stringop-overflow
+  CPPFLAGS += -DRISCV64 -Wno-error=stringop-truncation -Wno-error=stringop-overflow
```

Comment:
This is because the latest version of GNU cross toolchain comes up with GCC 9.2.0 which turns all warnings with `stringop-truncation `and `stringop-overflow` (they were introduced since GCC 8.0) into errors:
e.g.
```
jvminitcommon.c: In function 'findDllLoadInfo':
jvminitcommon.c:38:2: error: 'strncpy' 
specified bound 64 equals destination size 
[-Werror=stringop-truncation]
  strncpy(key, dllName, SMALL_STRING_BUF_SIZE);

riscv64-unknown-linux-gnu-g++ -DOPENJ9_BUILD -O3 -fno-strict-aliasing ...
 -DUT_DIRECT_TRACE_REGISTRATION -DTR_HOST_RISCV -c OSCache.cpp
OSCache.cpp: In static member function 'static IDATA 
SH_OSCache::removeCacheVersionAndGen(char*, UDATA, UDATA, const char*)':
OSCache.cpp:167:9: error: 'char* strncpy(char*, const char*, size_t)' 
specified bound depends on the length of the source argument 
[-Werror=stringop-overflow=]
  strncpy(buffer, nameStart, lengthMinusGen);
  ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
To make it clear and keep consistent with the change on OMR at #4433, I updated the script to exclude these two warnings only when compiling with GCC > 7 (e.g. if the build is compiled inside Fedora_riscv/QEMU in which only GCC 7.3.0 is offered).

Reply:
I suspect we could instead 'fix' the code so it doesn't elicit this warning; that would be preferable.

Reply:
> I suspect we could instead 'fix' the code so it doesn't elicit this warning; that would be preferable.

There are a bunch of code cross OpenJ9 and OMR ending up with the same failures if compiled with GCC > 8.0. I don't think it is necessary for RISC-V as GCC 7.3 is now used on Linux-based platforms.  'Fixing' these warnings will impact all platforms which is not the intention here.

@DanHeidinga, any input?

Reply:
@ChengJin01's answer makes sense - we should disable this warning for now and open an issue to track fixing it across the code base.  I don't think we need to hold up the RISC-V merges for a long standing issue.

For any new RISC-V code that adds this warning, we should address it immediately rather than making the issue worse

Reply:
These warnings occur on the existing code with `strncpy` which has nothing to with our changes.